CHANGELOG.md

```diff
@@ -1,3 +1,16 @@
+## 0.5.12-dev4
+
+### Enhancements
+
+* Added method to utils to allow date time format validation
+
+### Features
+
+* Add Slack connector to pull messages for a specific channel
+
+### Fixes
+
+
```

Comment:
note on procedure: when it is a -devX bump, just add the bullet points to the existing -dev section (in this case under 0.5.12-dev3) , then bump the devX version (so -dev3 becomes -dev4).